### PR TITLE
build(IDE): Support CodeBlocks no longer stand-alone, but only through CMake

### PR DESCRIPTION
**Pull request by quyykk**
This removes the CodeBlocks project in favor of the CB project that CMake can generate (also I am obligated to mention how much better VSCode is).

The instructions for setting up a CodeBlocks environment (through CMake) can still be found [on the master branch](https://github.com/endless-sky/endless-sky/blob/2c695adcd22d620dff4cdd03c7e62f0564903c13/readme-cmake.md) or [in the branch behind this PR](https://github.com/quyykk/endless-sky/blob/bye-codeblocks/readme-cmake.md#building-with-codeblocks)

### CODE REVIEW

**Comment by Saugia** [comment]
:(

File: README.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ Endless Sky has very minimal system requirements, meaning most systems should be
 |Storage Free | 250 MB | 1 GB |
 
 ## Building from source
-Most development is done on Linux and Windows, using the [SCons](https://scons.org/) build tool to compile the project. For those wishing to use an IDE, project files are provided for [Code::Blocks](https://www.codeblocks.org/) to simplify the project setup. There is also a CMake project available ([build instructions](readme-cmake.md)).
+Most development is done on Linux and Windows, using the [SCons](https://scons.org/) build tool to compile the project. There is also a CMake project available ([build instructions](readme-cmake.md)).
```
Comment (RisingLeaf) [inline README.md:24]: hopefully we will have a bye scons PR soon XD

**Comment by petervdmeer** [comment]
The people that care most about this PR are the people that use CodeBlocks, so I added a link for if such people start searching the PR list later to see what happened with the support to their much beloved IDE.

I also updated the title to match the "conventional commits" standard that we follow in ES, and added some relevant information to the tile.

**Review by petervdmeer — APPROVED** [review]

**Comment by petervdmeer** [comment]
@quyykk What is the status of this PR? It doesn't seem merge-able in its current state.

**Comment by ziproot** [comment]
Will https://cmake.org/cmake/help/latest/generator/CodeBlocks.html saying "deprecated since version 3.27" impact this PR? It looks like the ability to generate CodeBlocks projects through CMake as explained in the documentation might be impacted. Note that I do not use CodeBlocks so I don't care one way or another but if there are any CodeBlocks users, they might care about this.

**Comment by quyykk** [comment]
Good find! That's really a shame :pensive: Hopefully CodeBlocks will switch to the new API. I don't think it should impact this PR, anybody who still needs to use CodeBlocks can still use older CMake versions until CodeBlocks implements the newer CMake API.

**Comment by ziproot** [comment]
Fair enough.

**Comment by quyykk** [comment]
> Most other IDEs (e.g. VSCode, Visual Studio, XCode, CLion, Eclipse CDT, NetBeans, Sublime Text, Vim, Emacs, KDevelop, QT Creator) are not impacted by this.

Some of those are impacted too, everyone listed under https://cmake.org/cmake/help/latest/manual/cmake-generators.7.html#id16

**Comment by ziproot** [comment]
At first it looked like only CodeBlocks appears to still need it.
[CodeLite](https://docs.codelite.org/plugins/cmake/)
[Eclipse CDT](https://marketplace.eclipse.org/content/complete-eclipse-cc-ide)
[Sublime Text](https://packagecontrol.io/packages/CMakeBuilder)
[Kate](https://docs.kde.org/stable5/en/kate/kate/kate-application-plugin-build.html)
However, if people are using Sublime Text 2 they would have to pay to upgrade as the plugin only supports Sublime Text 3, so Sublime Text 2 users that do not also have Sublime Text 3 are also affected. Additionally, people relying on extra generators would have to change to use the plugins instead, so they are impacted. I apologize for the misleading statement above.

**Comment by quyykk** [comment]
Oh I didn't realize they all switched, except CodeBlocks. My bad. Hopefully they'll work on it then :smile:

**Comment by ziproot** [comment]
So far, it appears like the Code::Blocks developers don't care about extra generators being deprecated.
> Since they never really supported native CB Projects. Them dropping support for CB Custom makefile projects are not a major loss.

> From my point of view, I agree with you.
>
>The generated cbp(from CMakefiles) is just a makefile, and not a real CB native Project file.
See https://forums.codeblocks.org/index.php/topic,25536.0.html

The Code::Blocks developers are essentially saying that requiring CMake extra generators means you aren't supporting Code::Blocks. I think that you should reconsider this.

**Comment by quyykk** [comment]
> EDIT: There is a ticket open at [sourceforge.net/p/codeblocks/tickets/1428](https://sourceforge.net/p/codeblocks/tickets/1428/), so someone may add it.

That would be the best case scenario. The reason to remove the CB project is so that we can simplify the existing build/release process, and imo I would still be in favor of dropping it. I'll let others weigh in though :smile:

**Comment by ziproot** [comment]
After more people commented, I also have a medium case scenario that would still let this PR go through:
-The plugin "CBProjectExporter" currently lets you export a Code::Blocks project to CMake, but you will need to manually edit the file. Adding full support for exporting and importing would also work.

**Comment by petervdmeer** [comment]
> After more people commented, I also have a medium case scenario that would still let this PR go through: -The plugin "[CBProjectExporter](https://github.com/josephch/CBProjectExporter)" currently lets you export a Code::Blocks project to CMake, but you will need to manually edit the file. Adding full support for exporting and importing would also work. EDIT: Added link

Great!

**Comment by petervdmeer** [comment]
> The reason to remove the CB project is so that we can simplify the existing build/release process, and imo I would still be in favor of dropping it. I'll let others weigh in though 😄

I agree.

**Comment by xX-Dillinger-Xx** [comment]
Someone should update this  [BuildInstructions wiki](https://github.com/endless-sky/endless-sky/wiki/BuildInstructions/08483edf1a0373461a25b35d2dfa94b015f1150b)

**Comment by warp-core** [comment]
> Someone should update this [BuildInstructions wiki](https://github.com/endless-sky/endless-sky/wiki/BuildInstructions/08483edf1a0373461a25b35d2dfa94b015f1150b) It's not listed in the wiki but shows up with a google search

This is a link to an older version of the file. I'm not sure what we can do other than "give feedback" on the Google result and explain that the result links to an old version of a file that contains out of date information. Only the latest version of the file should be included in search results.

> https://github.com/endless-sky/endless-sky/wiki/BuildInstructions This also shows up but is not valid, it results in a 404 code

Thanks. I've updated the link in that file to point to the correct location.

**Comment by xX-Dillinger-Xx** [comment]
>This is a link to an older version of the file. 

I'm sure you already tried, but I'm confused. 
Can't you just delete it? If not just edit it to point to the new information.

**Pull request merged**